elfi/examples/daycare.py

```diff
@@ -0,0 +1,301 @@
+    equal number of sampled individuals among the daycare centers.
+
+    The model is defined as a continuous-time Markov process with transition probabilities:
+
+    Pr(I_{is}(t+dt)=1 | I_{is}(t)=0) = t1 * E_s(I(t)) + t2 * P_s, if \sum_{j=1}^N_s I_{ij}(t)=0
+    Pr(I_{is}(t+dt)=1 | I_{is}(t)=0) = t3 * (t1 * E_s(I(t)) + t2 * P_s), otherwise
+    Pr(I_{is}(t+dt)=0 | I_{is}(t)=1) = \gamma
+
+    where:
+    I_{is}(t) is the status of carriage of strain s for individual i.
+    E_s(I(t)) is the probability of sampling the strain s
+    t1 is the rate of transmission from other children at the DCC (\beta in paper).
+    t2 is the rate of transmission from the community outside the DCC (\Lambda in paper).
+    t3 scales the rate of an infected child being infected with another strain (\theta in paper).
+    \gamma is the relative probability of healing from a strain.
+
+    As in the paper, \gamma=1, and the other inferred parameters are relative to it.
+
+    The system is solved using the Direct method [Gillespie, 1977].
+
+    References
+    ----------
+    Numminen, E., Cheng, L., Gyllenberg, M. and Corander, J. (2013) Estimating the transmission
+        dynamics of Streptococcus pneumoniae from strain prevalence data, Biometrics, 69, 748-757.
+    Gillespie, D. T. (1977) Exact stochastic simulation of coupled chemical reactions.
+        The Journal of Physical Chemistry 81 (25), 2340–2361.
+
+    Parameters
+    ----------
+    t1 : float or np.array
+        Rate of transmission from other individuals at the DCC.
+    t2 : float or np.array
+        Rate of transmission from the community outside the DCC.
+    t3 : float or np.array
+        Scaling of co-infection for individuals infected with another strain.
+    n_ind : int, optional
+        Number of individuals in a DCC (same for all).
+    n_strains : int, optional
+        Number of bacterial strains considered.
+    freq_strains_commun : np.array of shape (n_strains,), optional
+        Prevalence of each strain in the community outside the DCC.
+    n_obs : int, optional
+        Number of individuals sampled from each DCC (same for all).
+    time_end : float, optional
+        The system is solved using the Direct method until all cases within the batch exceed this.
+    batch_size : int, optional
+    random_state : np.random.RandomState, optional
+
+    Returns
+    -------
+    state_obs : np.array
+        Observations in shape (batch_size, n_dcc, n_obs, n_strains).
+
+    """
+    random_state = random_state or np.random
+
+    t1 = np.asanyarray(t1).reshape((-1, 1, 1, 1))
+    t2 = np.asanyarray(t2).reshape((-1, 1, 1, 1))
+    t3 = np.asanyarray(t3).reshape((-1, 1, 1, 1))
+
+    if freq_strains_commun is None:
+        freq_strains_commun = np.full(n_strains, 0.1)
+
+    prob_commun = t2 * freq_strains_commun
+
+    state = np.zeros((batch_size, n_dcc, n_ind, n_strains), dtype=np.bool)  # infection status
```

Comment:
Could it be explicitly stated what the dimensions of the state are as the tensor-implementation may be difficult to follow.

Reply:
Added some explanation (including missing `n_dcc` in docstring).